### DISCUSSION TITLE
Access to spatial mesh?

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#334] [Q&A]
Is there a way to get to the actual geometry of the HoloLens spatial mesh? I looked a bit in the code and I don't it's exposed. Use case would be to export it in real-time over a network connection.

**Comment by maluoi** [comment]
I don't currently have that available! Microsoft is the only one with a Scene Understanding extension available, and I have _no_ idea what that's likely to look like in the future with devices that have no depth sensor. So I've been a little reluctant to surface too much information there, and make promises that I won't be able to keep in the long run.

I _probably_ should just do it anyhow though, so I'll think about it. Probably won't fit it into this release though, unless this is really high-priority for you. You could also directly work with the Scene Understanding EXT via `Backend.OpenXR` (like [FB passthrough](https://github.com/maluoi/StereoKit/blob/develop/Examples/StereoKitTest/Tools/PassthroughFBExt.cs)), it'll be a handful of work, but it does have a _lot_ of features that SK doesn't even hint at yet!
